src/main/java/algorithms/orthogonalRangeSearching/twoDim/OrthogonalRangeSearching.java

```diff
@@ -0,0 +1,314 @@
+            }
+        }
+    }
+
+    /**
+     * Performs a right traversal of the Y-tree to find points within the specified range.
+     *
+     * @param v      The current node in the Y-tree.
+     * @param high    The upper bound of the Y-coordinate range.
+     * @param result A list to store the results.
+     */
+    public static void rightTraversal(RangeTreeNode<Integer[]> v, int high, List<Integer[]> result) {
+        if (v != null) {
+            if (v.getLeft() == null && v.getRight() == null && v.getVal()[1] <= high) { // leaf, need extra check
+                result.add(v.getVal());
+            } else {
+                if (high > v.getVal()[1]) {
+                    allLeafTraversal(v.getLeft(), result);
+                    rightTraversal(v.getRight(), high, result);
+                } else { // a qualifying leaf might or might not exist, we are just exploring
+                    rightTraversal(v.getLeft(), high, result);
+                }
+            }
+        }
+    }
+
+    /**
+     * Searches for 2D points within the specified Y-coordinate range in the Y-tree.
+     *
+     * @param v      The root node of the Y-tree.
+     * @param yLow   The lower bound of the Y-coordinate range.
+     * @param yHigh  The upper bound of the Y-coordinate range.
+     * @param result A list to store the results.
+     */
+    public static void YSearch(RangeTreeNode<Integer[]> v, int yLow, int yHigh, ArrayList<Integer[]> result) {
+        if (v != null) {
+            System.out.println(v.getVal()[0]);
+            RangeTreeNode<Integer[]> splitNodeY = findYSplit(v.getYTree(), yLow, yHigh);
+            if (splitNodeY != null) {
+                if (splitNodeY.getLeft() == null && splitNodeY.getRight() == null
+                        && splitNodeY.getVal()[1] >= yLow && splitNodeY.getVal()[1] <= yHigh) { // if split node is leaf
+                    result.add(splitNodeY.getVal());
+                }
+                leftTraversal(splitNodeY.getLeft(), yLow, result);
+                rightTraversal(splitNodeY.getRight(), yHigh, result);
+            }
+        }
+    }
+
+    /**
+     * Searches for 2D points within the specified orthogonal range in the X-tree.
+     *
+     * @param tree   The root node of the X-tree.
+     * @param xLow   The lower bound of the X-coordinate range.
+     * @param xHigh  The upper bound of the X-coordinate range.
+     * @param yLow   The lower bound of the Y-coordinate range.
+     * @param yHigh  The upper bound of the Y-coordinate range.
+     * @return A list of 2D points within the specified orthogonal range.
+     */
+    public static List<Integer[]> search(RangeTreeNode<Integer[]> tree, int xLow, int xHigh, int yLow, int yHigh) {
+        RangeTreeNode<Integer[]> splitNodeX = findXSplit(tree, xLow, xHigh);
+        ArrayList<Integer[]> result = new ArrayList<>();
+        if (splitNodeX != null) {
+            if (splitNodeX.getLeft() == null && splitNodeX.getRight() == null
```

Comment:
slightly confused here.. if split node is a leaf, then the y-tree at this node has only 1 node no?

Reply:
yup, but still need to add the leaf into the result array and we abstracting that to the YSearch function

Reply:
ohhh